connection.cpp

```diff
@@ -176,6 +174,16 @@ void Connection::sync(int timeout)
     });
 }
 
+void Connection::onSyncSuccess(SyncData &data) {
```

Comment:
`onSyncSuccess()` should receive `SyncData&&`, to indicate that it consumes the structure.